tree_sitter_grep_filter_before_line_number/src/lib.rs

```diff
@@ -0,0 +1,25 @@
+use std::{
+    ffi::CStr,
+    sync::atomic::{AtomicUsize, Ordering},
+};
+
+use libc::c_char;
+use tree_sitter::Node;
+
+static ROW_NUMBER: AtomicUsize = AtomicUsize::new(0);
+
+#[no_mangle]
+pub extern "C" fn initialize(value: *const c_char) {
+    assert!(!value.is_null(), "Expected filter argument");
+    let value: usize = unsafe { CStr::from_ptr(value) }
```

Comment:
This is per eg [this](http://jakegoulding.com/rust-ffi-omnibus/string_arguments/) (for how to accept a string argument via FFI)

So the "ownership contract" here would seem to be that the plugin can't assume that this passed C string outlives the `initialize()` call (ie it doesn't own the passed string)? And here we shouldn't be violating that since we convert to an (owned) `usize`

Reply:
Ah oh, that makes sense